gst-plugin-pravega/src/pravegasink/imp.rs

```diff
@@ -73,11 +78,94 @@ pub enum TimestampMode {
     Tai = 2,
 }
 
+#[derive(Debug)]
+enum Retention {
+    Days(f64),
+    Bytes(u64),
+    DaysAndBytes(f64, u64),
+    None,
+}
+
+impl Retention {
+    fn new(retention_type: Option<String>, days: f64, bytes: u64) -> Self {
+        match retention_type {
+            Some(t) => {
+                match &t[..] {
+                    "days" => Self::Days(days),
+                    "bytes" => Self::Bytes(bytes),
+                    "daysAndBytes" => Self::DaysAndBytes(days, bytes),
+                    _ => Self::None,
+                }
+            }
+            _ => Self::None,
+        }
+    }
+}
+
+struct RetentionMaintainer {
+    retention: Retention,
+    factory: ClientFactory,
+    index_searcher: IndexSearcher<ByteReader>,
+    index_writer: ByteWriter,
+    data_writer: ByteWriter,
+}
+
+impl RetentionMaintainer {
+    fn new(retention: Retention, factory: ClientFactory, index_scoped_segment: ScopedSegment, data_scoped_segment: ScopedSegment) -> Self {
+        let index_reader = factory.create_byte_reader(index_scoped_segment.clone());
+        let index_writer = factory.create_byte_writer(index_scoped_segment);
+        let data_writer = factory.create_byte_writer(data_scoped_segment);
+        let index_searcher = IndexSearcher::new(index_reader);
+        Self {
+            retention,
+            factory,
+            index_searcher, 
+            index_writer,
+            data_writer,
+        }
+    }
+
+    fn run(mut self) {
+        let (days, bytes) = match self.retention {
+            Retention::Days(days) => (Some(days), None),
+            Retention::Bytes(bytes) => (None, Some(bytes)),
+            Retention::DaysAndBytes(days, bytes) => (Some(days), Some(bytes)),
+            _ => (None, None),
+        };
+        let seconds = if let Some(value) = days {
+            let sec = value * 24.0 * 60.0 * 60.0;
+            let sec = sec.round() as i128;
+            Some(sec)
+        } else {
+            None
+        };
```

Comment:
This function has two distinct parts that only share `seconds` between this selected portion and the next one. It would seem appropriate to refactor them into their own functions.

Reply:
The idea is if second is some, truncate by seconds, if bytes is some, truncate by bytes again. will add the truncate by bytes login when i implement the index_search by bytes in another PR.